File: fc-chat/demo/part3/ch3/05-1-message-system-add-channel/src/main/java/net/prostars/messagesystem/dto/websocket/outbound/FetchChannelInviteCodeResponse.java
package net.prostars.messagesystem.dto.websocket.outbound;

import net.prostars.messagesystem.constant.MessageType;
import net.prostars.messagesystem.dto.domain.ChannelId;
import net.prostars.messagesystem.dto.domain.InviteCode;

public class FetchChannelInviteCodeResponse extends BaseMessage {

  private final ChannelId channelId;
  private final InviteCode inviteCode;

  public FetchChannelInviteCodeResponse(ChannelId channelId, InviteCode inviteCode) {
    super(MessageType.FETCH_CHANNEL_INVITECODE_RESPONSE);
    this.channelId = channelId;
    this.inviteCode = inviteCode;
  }

  public ChannelId getChannelId() {
    return channelId;
  }

  public InviteCode getInviteCode() {
    return inviteCode;
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/dto/websocket/outbound/FetchChannelInviteCodeResponse.kt
package com.yscorp.withpush.messagesystem.dto.websocket.outbound

import com.yscorp.withpush.messagesystem.constant.MessageType
import com.yscorp.withpush.messagesystem.dto.domain.ChannelId
import com.yscorp.withpush.messagesystem.dto.domain.InviteCode


class FetchChannelInviteCodeResponse(val channelId: ChannelId, val inviteCode: InviteCode) :
    BaseMessage(MessageType.FETCH_CHANNEL_INVITECODE_RESPONSE) {
}
